src/checkpoints/checkpoints.service.ts

```diff
@@ -83,17 +83,21 @@ export class CheckpointsService {
     ticker: string,
     signerDto: TransactionBaseDto
   ): ServiceReturn<Checkpoint> {
-    const { signer, webhookUrl } = signerDto;
+    const { signer, webhookUrl, dryRun } = signerDto;
```

Comment:
I probably should have looked harder when doing webhooks. It seems like a smell to have to change every service to add a new base transaction parameter.

I tried out a generic helper and came up with this:

```ts
export const extractTxBase = <T extends TransactionBaseDto>(
  params: T
): {
  base: TransactionBaseDto;
  args: Omit<T, keyof TransactionBaseDto>;
} => {
  const { signer, webhookUrl, dryRun, ...args } = params;
  return {
    base: { signer, webhookUrl, dryRun },
    args,
  };
};
```

Used like:

```ts
  public async createAsset(params: CreateAssetDto): ServiceReturn<Asset> {
    const { base, args } = extractTxBase(params);

    const createAsset = this.polymeshService.polymeshApi.assets.createAsset;
    return this.transactionsService.submit(createAsset, args, base);
  }
```

Any thoughts on this? I think it will clean up the services a bit and going forward we shouldn't need to change every service for a new param.